flashinfer/norm/__init__.py

```
@@ -761,6 +761,588 @@ def fused_rmsnorm_silu(
+        raise ValueError(f"{name} must be 1D or 2D, got {tensor.ndim}D")
+    if tensor.shape[-1] != input.shape[2]:
+        raise ValueError(
+            f"{name} last dim must match hidden_dim {input.shape[2]}, "
+            f"got {tensor.shape[-1]}"
+        )
+    if tensor.dtype != torch.float32:
+        raise ValueError(f"{name} must be float32, got {tensor.dtype}")
+
+
+def _dit_ln_prepare_outputs(
+    input: torch.Tensor,
```

Ruff (0.15.11)

[error] Function argument `input` is shadowing a Python builtin

(A002)